apps/web/src/components/Posts/SharePostModal.tsx

```diff
@@ -0,0 +1,251 @@
+import { useToastActions } from '~/contexts/toast/ToastContext';
+import { SharePostModalQuery } from '~/generated/SharePostModalQuery.graphql';
+import { useIsMobileWindowWidth } from '~/hooks/useWindowSize';
+import FarcasterIcon from '~/icons/FarcasterIcon';
+import LensIcon from '~/icons/LensIcon';
+import TwitterIcon from '~/icons/TwitterIcon';
+import { contexts } from '~/shared/analytics/constants';
+import { getPreviewImageUrlsInlineDangerously } from '~/shared/relay/getPreviewImageUrlsInlineDangerously';
+import { noop } from '~/shared/utils/noop';
+import { getBaseUrl } from '~/utils/getBaseUrl';
+
+import breakpoints from '../core/breakpoints';
+import { Button } from '../core/Button/Button';
+import { HStack, VStack } from '../core/Spacer/Stack';
+import { MiniPostOpenGraphPreview } from './MiniPostOpenGraphPreview';
+
+type Props = {
+  postId: string;
+  tokenName?: string;
+};
+
+export default function SharePostModal({ postId, tokenName = 'this' }: Props) {
+  const queryResponse = useLazyLoadQuery<SharePostModalQuery>(
+    graphql`
+      query SharePostModalQuery($postId: DBID!) {
+        post: postById(id: $postId) {
+          ... on ErrPostNotFound {
+            __typename
+          }
+          ... on Post {
+            __typename
+            author @required(action: THROW) {
+              username
+              profileImage {
+                ... on TokenProfileImage {
+                  token {
+                    ...getPreviewImageUrlsInlineDangerouslyFragment
+                  }
+                }
+                ... on EnsProfileImage {
+                  __typename
+                  profileImage {
+                    __typename
+                    previewURLs {
+                      medium
+                    }
+                  }
+                }
+              }
+            }
+            caption
+            tokens {
+              ...getPreviewImageUrlsInlineDangerouslyFragment
+            }
+          }
+        }
+      }
+    `,
+    { postId: postId }
+  );
+
+  const { post } = queryResponse;
+  const postUrl = `${getBaseUrl()}/post/${postId}`;
+
+  const { pushToast } = useToastActions();
+
+  const handleCopyToClipboard = useCallback(async () => {
+    void navigator.clipboard.writeText(postUrl);
+    pushToast({ message: 'Copied link to clipboard', autoClose: true });
+  }, [postUrl, pushToast]);
+
+  const handleShareButtonClick = useCallback(
+    (baseComposePostUrl: string) => {
+      const message = `I just posted ${tokenName} on gallery ${postUrl}`;
+      const encodedMessage = encodeURIComponent(message);
+
+      window.open(`${baseComposePostUrl}?text=${encodedMessage}`, '_blank');
+    },
+    [tokenName, postUrl]
+  );
+
+  // stripped down version of the pfp retrieving logic in ProfilePicture.tsx
+  const profileImageUrl = useMemo(() => {
+    if (!post || post?.__typename !== 'Post') {
+      return null;
+    }
```

Comment:
nit: should be able to toss the left side and use `if (post?.__typename !== 'Post')`